src/java.naming/share/classes/com/sun/naming/internal/VersionHelper.java

```diff
@@ -217,20 +158,16 @@ InputStream getResourceAsStream(Class<?> c, String name) {
      *
      * @param filename  The file name, sans directory.
      */
-    @SuppressWarnings("removal")
     InputStream getJavaHomeConfStream(String filename) {
-        PrivilegedAction<InputStream> act = () -> {
-            try {
-                String javahome = System.getProperty("java.home");
-                if (javahome == null) {
-                    return null;
-                }
-                return Files.newInputStream(Path.of(javahome, "conf", filename));
-            } catch (Exception e) {
+        try {
+            String javahome = System.getProperty("java.home");
```

Comment:
StaticProperty.javaHome() is available as a stable value for the property.

Reply:
Good point too - but that would require am additional qualified export of `jdk.internal.util` from `java.base` to `java.naming`. I'm not sure it's worth it. I mean - keeping qualified exports as few as possible is also a worthy goal.

Reply:
Never mind, I forgot this isn't in java.base.